Add custom event data template to imap integration (#27501)
home-assistant/home-assistant.github.io: source/_integrations/imap.markdown

When a new message arrives that meets the search criteria the `imap` integration will send a custom [event](/docs/automation/trigger/#event-trigger) that can be used to trigger an automation.
It is also possible to use to create a template [`binary_sensor` or `sensor`](/integrations/template/#trigger-based-template-binary-sensors-buttons-numbers-selects-and-sensors) based the [event data](/docs/automation/templating/#event).

The table below shows what attributes come with `trigger.event.data`. The data is a dictionary that has the keys that are shown below.

> The attributes shown in table are also available as variables for the custom event data template. The [example](/integrations/imap/#example---custom-event-data-template) shows how to use this as an event filter.

Reviewer: ```suggestion
The attributes shown in the table are also available as variables for the custom event data template. The [example](/integrations/imap/#example---custom-event-data-template) shows how to use this as an event filter.
```

<div class='note info'>

> The custom event data template is an advanced feature. The option is available only when advanced mode is enabled (see user settings). The `text` attribute is not size limited when it is used as variable in the template.

Reviewer: ```suggestion
The custom event data template is an advanced feature. The option is available only when advanced mode is enabled (see user settings). The `text` attribute is not size limited when used as a variable in the template.
```

{% endraw %}

By making small changes to the regular expressions defined above, a similar structure can parse other types of data out of the body text of other emails.

## Example - custom event data template

> We can define a custom event data template to help filtering events. This can be handy if e.g. we have multiple senders we want to allow.

Reviewer: ```suggestion
We can define a custom event data template to help filter events. This can be handy if, for example, we have multiple senders we want to allow.
```
